Geirst/fix disk and memory usage sampling for docker

geirst opened the pull request:
@toregge and @baldersheim please review

geirst committed:
Extend HwInfo with disk size and memory size and sample these if not specified in config.

geirst committed:
Use HwInfo for disk and memory size in DiskMemUsageSampler and sample used disk size under vespa home dir.

This should fix sampling of disk usage when using docker.

toregge reviewed (approved):
LGTM.

baldersheim commented on searchcore/src/vespa/searchcore/proton/server/disk_mem_usage_sampler.cpp:
```diff
@@ -59,10 +45,26 @@ DiskMemUsageSampler::sampleUsage()
 {
-    _filter.setDiskStats(std::experimental::filesystem::space(_path));
+    _filter.setDiskStats(sampleDiskUsageInDirectory(_path));
 }
```
Check if disk is slow or fast

baldersheim commented:
Fine, except from the comment.

geirst committed:
Sample disk usage on file system (as before) if disk is slow.

This should avoid added cost of scanning directory structure in slow-disk case.

geirst commented:
@baldersheim please review again

baldersheim reviewed (approved):
👍
